src/stan/callbacks/json_writer.hpp

```diff
@@ -109,7 +97,17 @@ class json_writer {
    * @param[in] key member name.
    */
   void write_key(const std::string& key) {
-    *output_ << "\"" << process_string(key) << "\" : ";
+    *output_ << std::string(record_depth_ * 2, ' ') << "\""
+             << process_string(key) << "\" : ";
+  }
+
+  template <typename T>
+  void write_int_like(const std::string& key, T value) {
+    if (output_ == nullptr)
+      return;
+    write_sep();
+    write_key(key);
+    *output_ << value;
   }
```

Comment:
I would just call this `write_` since it takes anything that is valid for `operator<<`

Reply:
Oh for NA etc.

I'd still call it `write_` and have the signature like 

```cpp
template <typename T, stan::require_integral_t<T>* = nullptr>
  void write_int_like(const std::string& key, T value) 
```